src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/StackExchangeRedis2Plus/SessionCache.cs

```diff
@@ -71,6 +75,28 @@ public void Harvest(ISegment hostSegment)
             }
         }
 
+        private void CleanUp()
+        {
+            var cleanedSessions = 0;
+            foreach (var pair in _sessionCache)
+            {
+                lock (pair.Key)
+                {
+                    if (pair.Key.IsDone
+                        || !(pair.Value.transaction?.TryGetTarget(out var txn) ?? false)
+                        || txn.IsFinished)
```

Comment:
We might want to move the transaction check outside of the lock. I don't think that we want to remove the segments if they are not done, even if the transaction is done. If we cleanup segments that are not done here, it may be the case that the transaction finalization process has not forcibly ended the child segments yet, which would prevent us from capturing the redis segments since this code does not try to capture the redis segments.